src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -79,39 +76,35 @@ def initialize_jobs(self):
         for job_id, job_state in job_states.items():
             job = Job.from_state(job_state)
             status = job_state.get("status")
-            refresh_state = 1 if status not in TERMINAL_STATES else 0
+            refresh_state = 1 if status not in TERMINAL_STATUSES else 0
             self._running_jobs[job_id] = {
                 "refresh": refresh_state,
                 "job": job,
             }
-            # if the job is in a terminal state, add it to the _completed_job_states index
-            if not refresh_state:
-                revised_state = self._construct_job_status(
-                    self.get_job(job_id), job_state
-                )
-                self._completed_job_states[job_id] = revised_state
 
-    def _create_jobs(self, job_ids):
+    def _create_jobs(self, job_ids, job_states=None):
         """
         TODO: error handling
         Given a list of job IDs, creates job objects for them and populates the _running_jobs dictionary
         """
-        job_states = clients.get("execution_engine2").check_jobs(
-            {
-                "job_ids": job_ids,
-                "return_list": 0,
-                "exclude_fields": JOB_INIT_EXCLUDED_JOB_STATE_FIELDS,
-            }
-        )
+        job_ids = [job_id for job_id in job_ids if job_id not in self._running_jobs]
+        if not job_states:
```

Comment:
do we need to check whether job states are present for all the jobs in `job_ids`?

Reply:
Yeah, unclear to me. At least it's what it was doing before